pkg/util/procfs/procfs.go

```diff
@@ -45,7 +46,15 @@ func containerNameFromProcCgroup(content string) (string, error) {
 // Eg. If the devices cgroup for the container is stored in /sys/fs/cgroup/devices/docker/nginx,
 // return docker/nginx. Assumes that the process is part of exactly one cgroup hierarchy.
 func (pfs *ProcFs) GetFullContainerName(pid int) (string, error) {
-	filePath := path.Join("/proc", strconv.Itoa(pid), "cgroup")
+	var filePath string
+	// if kubelet was started with host's rootfs mounted, assume it's running as a container
```

Comment:
Even if kubelet is running in a container, AFAIK it currently assumes it has control over the entire node. I hope those assumptions are not an issue for your use case.